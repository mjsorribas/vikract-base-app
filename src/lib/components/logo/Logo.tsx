import { Image, Box, Text } from "@chakra-ui/react";

interface Props {
  boxSize?: string;
  fontSize?: string;
  fontColor?: string;
  design?: string;
  img?: string;
  imgTitle?: string;
  title?: string;
}

const defaultProps: Props = {
  boxSize: "60px",
  fontSize: "2rem",
  fontColor: "#000000",
  design: "full",
  title: "THE VIKRACT TEMPLATE",
  img: "/assets/vikract.svg",
  imgTitle: "The Vikract Webapp template",
};

const displayImg = ({
  boxSize,
  img,
  imgTitle,
}: Props & typeof defaultProps) => {
  return (
    <Image boxSize={boxSize} objectFit="cover" src={img} title={imgTitle} />
  );
};

const displayText = ({
  fontColor,
  fontSize,
  title,
}: Props & typeof defaultProps) => {
  return (
    <Text color={fontColor} fontSize={fontSize} fontWeight={300}>
      {title}
    </Text>
  );
};
const displayFull = ({
  boxSize,
  fontSize,
  fontColor,
  img,
  title,
  imgTitle,
}: Props & typeof defaultProps) => {
  return (
    <Box>
      <Image boxSize={boxSize} objectFit="cover" src={img} title={imgTitle} />
      <Text color={fontColor} fontSize={fontSize} fontWeight={300}>
        {title}
      </Text>
    </Box>
  );
};

const renderSwitch = (props: Props & typeof defaultProps) => {
  switch (props.design) {
    case "onlyImg":
      return displayImg(props);
    case "onlyText":
      return displayText(props);
    default:
      return displayFull(props);
  }
};

const Logo = (props = defaultProps) => {
  return (
    <Box display="flex" alignItems="center">
      {renderSwitch(props)}
    </Box>
  );
};

Logo.defaultProps = defaultProps;

export default Logo;
