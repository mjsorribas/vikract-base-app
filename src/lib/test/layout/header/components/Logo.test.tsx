import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Logo from "lib/components/logo/Logo";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test("Logo", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <Logo />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test("Logo onlyText", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <Logo design="onlyText" />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test("Logo onlyImg", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <Logo design="onlyImg" />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
