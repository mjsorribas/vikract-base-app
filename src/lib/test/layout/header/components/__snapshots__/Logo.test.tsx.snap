// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo 1`] = `
<div
  className="css-70qvj9"
>
  <div
    className="css-0"
  >
    <img
      className="chakra-image css-v1ux0r"
      src="/assets/vikract.svg"
      title="The Vikract Webapp template"
    />
    <p
      className="chakra-text css-j2414u"
    >
      THE VIKRACT TEMPLATE
    </p>
  </div>
</div>
`;

exports[`Logo onlyImg 1`] = `
<div
  className="css-70qvj9"
>
  <img
    className="chakra-image css-v1ux0r"
    src="/assets/vikract.svg"
    title="The Vikract Webapp template"
  />
</div>
`;

exports[`Logo onlyText 1`] = `
<div
  className="css-70qvj9"
>
  <p
    className="chakra-text css-j2414u"
  >
    THE VIKRACT TEMPLATE
  </p>
</div>
`;
